experimental/segmenter/src/lstm.rs

```diff
@@ -23,6 +23,7 @@ impl Iterator for LstmSegmenterIterator<'_> {
     type Item = usize;
 
     fn next(&mut self) -> Option<Self::Item> {
+        #[allow(clippy::indexing_slicing)] // pos_utf8 in range
```

Comment:
Nit (optional): Since you're already using `?` in this function, it would be simpler if you did

```
self.pos_utf8 += self.input.get(self.pos_utf8..)?.chars().next()?.len_utf8();
```